### DISCUSSION TITLE
Proper way to return tensor from node

### DISCUSSION THREAD

**Discussion by WASasquatch** [Comfy-Org/ComfyUI#124] [Q&A] [answered]
I am having issues converting a numpy to a proper tensor that doesn't result in 768 individual slices on the Y axis x.x

I am adding a MiDaS depth node to my custom nodes, but can't get a workable image for other nodes. I can save out the image just fine from PIL (below). 

Example:
```python
        depth = prediction.cpu().numpy()
        depth = (depth * 255 / (np.max(depth)+1)).astype('uint8')
        depth_image = Image.fromarray(depth)
        depth_image.save('TEST_DEPTH.png')
        tensor = torch.from_numpy( depth )
        tensors = ( tensor, )
```
![TEST_DEPTH](https://user-images.githubusercontent.com/1151589/225825206-c27dfb5c-8d29-4a1b-ad1a-5b4f4b4a6d4e.png)

**Comment by comfyanonymous** [comment, marked as answer]
See the LoadImage node: https://github.com/comfyanonymous/ComfyUI/blob/master/nodes.py#L863

**Reply by WASasquatch** [reply]
Oh gawd! Thank you! I was looking at like every other node (decoders and stuff) but that, which is silly in hindsight. Lol

**Comment by wwdok** [comment]
But sometime the tensor is not image, it can not be visualized simply like image

**Reply by WASasquatch** [reply]
You'd want to give a custom type to a tesnor that is not a image.
